biorbd_optim/integrator.py

```diff
@@ -31,6 +32,24 @@ def dxdt(h, states, controls, params):
             k3 = fun(x[:, i - 1] + h / 2 * k2, u, p)[:, idx]
             k4 = fun(x[:, i - 1] + h * k3, u, p)[:, idx]
             x[:, i] = x[:, i - 1] + h / 6 * (k1 + 2 * k2 + 2 * k3 + k4)
+
+        nb_dof = 0
+        quat_idx = []
+        quat_number = 0
+        for j in range(model.nbSegment()):
+            if model.segment(j).isRotationAQuaternion():
+                quat_idx.append([nb_dof, nb_dof + 1, nb_dof + 2, model.nbDof() + quat_number])
+                quat_number += 1
+            nb_dof += model.segment(j).nbDof()
+
+        for j in range(model.nbQuat()):
+            quaternion = vertcat(
+                x[quat_idx[j][3], i], x[quat_idx[j][0], i], x[quat_idx[j][1], i], x[quat_idx[j][2], i]
```

Comment:
Can you confirm what is this "i" for? it seems that it refer to 2 for loops before, but it is not included. Is that mean that i is supposed to be last value? If so please use "-1", otherwise it seems that indentation is wrong